indexing-service/src/main/java/io/druid/indexing/common/actions/SegmentAllocateAction.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.common.actions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
+import com.metamx.common.ISE;
+import io.druid.indexing.common.TaskLock;
+import io.druid.indexing.common.task.Task;
+import io.druid.segment.realtime.appenderator.SegmentIdentifier;
+import org.joda.time.Interval;
+
+import java.io.IOException;
+
+public class SegmentAllocateAction implements TaskAction<SegmentIdentifier>
+{
+  private final String dataSource;
+  private final Interval interval;
```

Comment:
didn't we want client to send a timestamp? e.g. kafka/tranquility task receives a [very] late event and has no idea of what segment granularity was used in the past, so the implementation needs to figure out correct interval given that dataSource and timestamp.